src/searching/binary_search.rs

```diff
@@ -0,0 +1,220 @@
+use std::cmp::Ordering;
+/// ## Binary Search algorithm.
+/// Iterating over the array by dividing half of the array and returns the index value of the target element in the array.
+/// you can pass both ascending and decending array respectively.
+///
+/// Time complexcity = O(logN) where `N` is the length of the array.
+///
+/// ## Arguments
+/// `array` - The array where you want to find.
+///
+/// `item` - The item which you want to find.
+///
+/// ## Returns
+/// `index` - the index value of that number if the item is in the array.
+///
+/// `None` - If the item is not found in the array.
+///
+
+pub fn binary_search<T: Ord>(array: &[T], item: &T) -> Option<usize> {
+    // assuming the given array is ascending.
+    let mut is_asc = true;
+
+    // checking if the given array contains more than 1 element or not because.
+    // like [2]. here there is no way to tell the elements are ascending or descending.
+    // like [1, 2] here you can check if the elements are ascending or descending.
```

Comment:
The wording here is not clear, partially because the implementation cares about whether the array is non-decreasing or not.